packages/purpose-process/test/testGetPurposes.ts

```diff
@@ -231,19 +231,47 @@ export const testGetPurposes = (): ReturnType<typeof describe> =>
           eservicesIds: [],
           consumersIds: [],
           producersIds: [],
-          states: [purposeVersionState.rejected, purposeVersionState.archived],
+          states: [purposeVersionState.rejected, purposeVersionState.active],
           excludeDraft: undefined,
         },
         { offset: 0, limit: 50 },
         genericLogger
       );
-      expect(result.totalCount).toBe(3);
+      expect(result.totalCount).toBe(4);
       expect(result.results).toEqual([
+        mockPurpose3,
         mockPurpose4,
         mockPurpose5,
         mockPurpose6,
       ]);
     });
+    it("should get the purposes with only archived versions (and exclude the ones with both archived and non-archived versions)", async () => {
```

Comment:
it could also be useful a separate test where both archived and non-archived state is passed to the filter

Reply:
Added with a [note](https://github.com/pagopa/interop-be-monorepo/pull/438/files#r1598198421)